### ISSUE TITLE
Disable default redirect of DNSLink websites if it breaks Origin

### ISSUE THREAD

**Issue by lidel** [ipfs-shipyard/ipfs-companion#701]
> Extracted from parent issue: https://github.com/ipfs-shipyard/ipfs-companion/issues/667, read it for full context.

> [..] with latest Beta ([v2.7.5.762](https://github.com/ipfs-shipyard/ipfs-companion/releases/tag/v2.7.5.762)) we have almost all the pieces of the UX for **( B )**:
> 
> - It enables people to opt-out from redirects per hostname. (#687)
> - It shows IPFS actions on DNSLink websites even when redirect is disabled (#694)
> 
> What is missing: **in case of DNSLink websites, we want to tweak this behaviour and disable redirect of DNSLink websites by default.** People will still be able to opt-in via <del>toggle introduced in (#687) and a global one in _Preferences_</del> _**new menu item**_ if they choose to do so, but the default should not change the Origin.

This will also solve the problem of websites hanging on initial load due to IPNS being slow ;-)

**Comment by lidel** [comment]
Note to self:

I feel it would be confusing if we introduced per-site opt-out in one Stable release, and then change the default behavior to not redirect DNSLink websites in another one. 

Makes sense to wait with Stable release until the above tweak is implemented.

**Comment by lidel** [comment]
After chats with @autonome, @hugomrdias and @cwaring we decided to bump the priority of this.

Implementation notes:

- IPFS context actions will continue to be available
- If HTTP server backing DNSLink website goes down, IPFS Companion will recover in a new tab automatically (#640, https://github.com/ipfs-shipyard/ipfs-companion/pull/783, https://github.com/ipfs-shipyard/ipfs-companion/pull/797) 
  - This is to emphasize there is no regression for offline and censorship scenarios
- There will be a new menu item for users who choose to break origin separation and load from local gateway.
  - In short, opt-in instead of opt-out, to respect Origin-based isolation.
